File: packages/typescript/src/endpoints/repositories.ts
/**
 * Repositories API Client
 */

import { AxiosInstance } from 'axios'

export class RepositoriesClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * POST /sync
   */
  async postSync(background_tasks: BackgroundTasks): Promise<List[RepositoryResponse]> {
        const params = {}
        const data = background_tasks
        return this.httpClient.post(`/sync`, data, params)
  }
  /**
   * GET /
   */
  async getRoot(connection_id?: string, language?: string, scan_enabled?: boolean, search?: string, page?: number, per_page?: number): Promise<RepositoryListResponse> {
        const params = {'connection_id: connection_id, language: language, scan_enabled: scan_enabled, search: search, page: page, per_page: per_page'}
        const data = undefined
        return this.httpClient.get(`/`, params)
  }
  /**
   * GET /{repository_id}
   */
  async get{repository_id}(repository_id: string): Promise<RepositoryResponse> {
        const params = {'repository_id: repository_id'}
        const data = undefined
        return this.httpClient.get(`/{repository_id}`, params)
  }
  /**
   * PUT /{repository_id}
   */
  async put{repository_id}(repository_id: string, repository_update: RepositoryUpdate): Promise<RepositoryResponse> {
        const params = {}
        const data = {'repository_id: repository_id, repository_update: repository_update'}
        return this.httpClient.put(`/{repository_id}`, data, params)
  }
  /**
   * DELETE /{repository_id}
   */
  async delete{repository_id}(repository_id: string): Promise<None> {
        const params = {}
        const data = repository_id
        return this.httpClient.delete(`/{repository_id}`, data, params)
  }
  /**
   * GET /{repository_id}/scans
   */
  async get{repository_id}scans(repository_id: string, limit?: number): Promise<List[ScanInDB]> {
        const params = {'repository_id: repository_id, limit: limit'}
        const data = undefined
        return this.httpClient.get(`/{repository_id}/scans`, params)
  }
  /**
   * POST /{repository_id}/scan
   */
  async post{repository_id}scan(repository_id: string, background_tasks: BackgroundTasks): Promise<ScanInDB> {
        const params = {}
        const data = {'repository_id: repository_id, background_tasks: background_tasks'}
        return this.httpClient.post(`/{repository_id}/scan`, data, params)
  }
  /**
   * GET /{repository_id}/branches
   */
  async get{repository_id}branches(repository_id: string): Promise<List[RepositoryBranchResponse]> {
        const params = {'repository_id: repository_id'}
        const data = undefined
        return this.httpClient.get(`/{repository_id}/branches`, params)
  }
  /**
   * POST /{repository_id}/pause
   */
  async post{repository_id}pause(repository_id: string): Promise<RepositoryResponse> {
        const params = {}
        const data = repository_id
        return this.httpClient.post(`/{repository_id}/pause`, data, params)
  }
  /**
   * POST /{repository_id}/resume
   */
  async post{repository_id}resume(repository_id: string): Promise<RepositoryResponse> {
        const params = {}
        const data = repository_id
        return this.httpClient.post(`/{repository_id}/resume`, data, params)
  }
  /**
   * GET /{repository_id}/analytics
   */
  async get{repository_id}analytics(repository_id: string, timeframe?: string): Promise<Dict[str, Any]> {
        const params = {'repository_id: repository_id, timeframe: timeframe'}
        const data = undefined
        return this.httpClient.get(`/{repository_id}/analytics`, params)
  }
  /**
   * GET /{repository_id}/badge
   */
  async get{repository_id}badge(repository_id: string, style?: string): Promise<Dict[str, Any]> {
        const params = {'repository_id: repository_id, style: style'}
        const data = undefined
        return this.httpClient.get(`/{repository_id}/badge`, params)
  }
  /**
   * GET /{repository_id}/activity
   */
  async get{repository_id}activity(repository_id: string, limit?: number): Promise<List[Dict[str, Any]]> {
        const params = {'repository_id: repository_id, limit: limit'}
        const data = undefined
        return this.httpClient.get(`/{repository_id}/activity`, params)
  }
}

File: packages/python/src/endpoints/repositories.py
"""
Repositories API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class RepositoriesClient:
    """Client for repositories endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def post_sync(self, background_tasks: Any):
        """POST /sync"""
        url = f"/sync"
        params = {}
        data = background_tasks
        return await self._request("POST", url, params=params, json=data)


    async def get_root(self, connection_id: Optional[str] = None, language: Optional[str] = None, scan_enabled: Optional[bool] = None, search: Optional[str] = None, page: Optional[float] = None, per_page: Optional[float] = None):
        """GET /"""
        url = f"/"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_repository_id(self, repository_id: str):
        """GET /{repository_id}"""
        url = f"/{repository_id}"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def put_repository_id(self, repository_id: str, repository_update: Any):
        """PUT /{repository_id}"""
        url = f"/{repository_id}"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("PUT", url, params=params, json=data)


    async def delete_repository_id(self, repository_id: str):
        """DELETE /{repository_id}"""
        url = f"/{repository_id}"
        params = {}
        data = repository_id
        return await self._request("DELETE", url, params=params, json=data)


    async def get_repository_id_scans(self, repository_id: str, limit: Optional[float] = None):
        """GET /{repository_id}/scans"""
        url = f"/{repository_id}/scans"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_repository_id_scan(self, repository_id: str, background_tasks: Any):
        """POST /{repository_id}/scan"""
        url = f"/{repository_id}/scan"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def get_repository_id_branches(self, repository_id: str):
        """GET /{repository_id}/branches"""
        url = f"/{repository_id}/branches"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def post_repository_id_pause(self, repository_id: str):
        """POST /{repository_id}/pause"""
        url = f"/{repository_id}/pause"
        params = {}
        data = repository_id
        return await self._request("POST", url, params=params, json=data)


    async def post_repository_id_resume(self, repository_id: str):
        """POST /{repository_id}/resume"""
        url = f"/{repository_id}/resume"
        params = {}
        data = repository_id
        return await self._request("POST", url, params=params, json=data)


    async def get_repository_id_analytics(self, repository_id: str, timeframe: Optional[str] = None):
        """GET /{repository_id}/analytics"""
        url = f"/{repository_id}/analytics"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_repository_id_badge(self, repository_id: str, style: Optional[str] = None):
        """GET /{repository_id}/badge"""
        url = f"/{repository_id}/badge"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_repository_id_activity(self, repository_id: str, limit: Optional[float] = None):
        """GET /{repository_id}/activity"""
        url = f"/{repository_id}/activity"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)

